### PR TITLE
feat: template view header

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### <samp>🤖 Generated by Copilot at d4d29ca</samp>

Enhanced the `SocialImage` component to support different sizes and styles, and updated the `TemplateView` component to use MUI components, custom hooks and a new header component. The header component displays the template metadata and the social image, and allows editing the template settings based on the user role.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/34356579/todos/6557488312)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] it should render template header with buttons
- [ ] it should render template header with buttons on mobile

# Walkthrough

### <samp>🤖 Generated by Copilot at d4d29ca</samp>

*  Add `TemplateViewHeader` component to render template metadata and social image (F6,F7)
  - Use `variant` prop of `SocialImage` component to display larger image on larger screens ([link](https://github.com/JesusFilm/core/pull/1953/files?diff=unified&w=0#diff-2f472b03d2f1e26b611957f1864d7870c6fcd7adfe64f39533e543a2485d0b6dL8-R14),[link](https://github.com/JesusFilm/core/pull/1953/files?diff=unified&w=0#diff-2f472b03d2f1e26b611957f1864d7870c6fcd7adfe64f39533e543a2485d0b6dL25-R32),[link](https://github.com/JesusFilm/core/pull/1953/files?diff=unified&w=0#diff-2f472b03d2f1e26b611957f1864d7870c6fcd7adfe64f39533e543a2485d0b6dL39-R48))
  - Replace `Box` component with `Stack` component for responsive layout ([link](https://github.com/JesusFilm/core/pull/1953/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150L1-R1))
  - Import and use `Role` enum, `useUserRoleQuery` hook, and `TemplateViewHeader` component ([link](https://github.com/JesusFilm/core/pull/1953/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150L8-R12))
  - Use `TemplateViewHeader` component to render template title, description, social image, create journey button and edit button ([link](https://github.com/JesusFilm/core/pull/1953/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150L18-R28))

### CODE REVIEW

File: apps/journeys-admin/src/components/TemplateView/TemplateView.tsx [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
   return (
     <Stack gap={4}>
-      <Typography variant="h1">{journey?.title}</Typography>
-      <Typography variant="body1">{journey?.description}</Typography>
-      <CreateJourneyButton signedIn={authUser?.id != null} />
+      <TemplateViewHeader isPublisher={isPublisher} authUser={authUser} />
+      <Typography variant="body2" sx={{ display: { xs: 'block', sm: 'none' } }}>
+        {journey?.description}
+      </Typography>
       <Stack sx={{ pt: 4 }}>
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateView.tsx:30]: remove this

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateView.tsx (hunk 1 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateView/TemplateView.tsx]: This is still in the codebase but moved below the template cards. This needs to be here, it's in the design

File: apps/journeys-admin/src/components/TemplateView/TemplateView.tsx [hunk 2]
```diff
@@ -1,31 +1,51 @@
-import Stack from '@mui/material/Box'
+import Stack from '@mui/material/Stack'
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateView/TemplateView.tsx:1]: not sure why this didn't in previously merged PR's...

File: apps/journeys-admin/src/components/TemplateView/TemplateView.tsx [hunk 3]
```diff
@@ -1,31 +1,51 @@
-import Stack from '@mui/material/Box'
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateView/TemplateView.tsx:1]: not sure why this didn't lint in previous PR's before it was merged... 😅

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+jest.mock('@mui/material/useMediaQuery', () => ({
+  __esModule: true,
+  default: () => true
+}))
+
+describe('TemplateViewHeader', () => {
+  const featuredJourney = { ...journey, featuredAt: '2023-10-12T08:00:00Z' }
+
+  it('should render the social image placeholder', () => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx:21]: dont need this test, should be already tested from the social share image itself

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        </SnackbarProvider>
+      </MockedProvider>
+    )
+
+    expect(getByText('October 2023')).toBeInTheDocument()
+    expect(getByText('my journey')).toBeInTheDocument()
+    expect(getByText('my cool journey')).toBeInTheDocument()
+  })
+
+  it('should render Use Template button for authenticated users', async () => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx:93]: Shouldn't this also show up for un authed users?

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateView.tsx [hunk 6]
```diff
@@ -1,31 +1,51 @@
...
     <Stack gap={4}>
-      <Typography variant="h1">{journey?.title}</Typography>
-      <Typography variant="body1">{journey?.description}</Typography>
-      <CreateJourneyButton signedIn={authUser?.id != null} />
-      <Stack sx={{ pt: 4 }}>
-        <TemplatePreviewTabs />
-      </Stack>
+      {journey?.featuredAt != null && (
+        <Typography
+          data-testId="featuredAtTemplatePreviewPage"
```
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateView/TemplateView.tsx:30]: this component is outside of the header in design

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+}: TemplateViewHeaderProps): ReactElement {
+  const { journey } = useJourney()
+  const smUp = useMediaQuery((theme: Theme) => theme.breakpoints.up('sm'))
+
+  return (
+    <Stack>
+      <Stack gap={4} sx={{ display: 'flex', flexDirection: 'row' }}>
+        <Box
+          sx={{
+            width: { xs: 107, sm: 244 },
```
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx:35]: need flex shrink 0 in this box, dropped height and width css props

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+          <Box
+            sx={{
+              display: { xs: 'none', sm: 'flex' },
+              gap: 4,
+              marginTop: 'auto'
+            }}
+          >
+            <CreateJourneyButton signedIn={authUser?.id != null} />
+            <PreviewTemplateButton slug={journey?.slug} />
+            {isPublisher != null && isPublisher && <TemplateEditButton />}
```
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx:87]: type error if done that way

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx (hunk 5 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx:93]: that's my bad, I named the test wrong

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx [hunk 9]
```diff
@@ -70,7 +55,7 @@ describe('TemplateViewHeader', () => {
     expect(getByRole('img', { name: 'image.jpg' })).toBeInTheDocument()
   })
 
-  it('should display the featured date, title, and description', () => {
+  it('should display the  title, and description', () => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx:58]: theres a double space in here 
```suggestion
  it('should display the title and description', () => {
```

**Review by csiyang — CHANGES_REQUESTED** [review]

**Review by csiyang — DISMISSED** [review]

**Review by csiyang — APPROVED** [review]

**Pull request merged**